### ISSUE TITLE
Mysql2::Error: Data too long for column 'output' at row 1

### ISSUE THREAD

**Issue by bernardinosousa** [postalserver/postal#1687]
## Describe the bug

Output of certains emails got spam detected and length of message being greater than 512, which couldn´t insert into postal.

## To Reproduce

Send an email and got spam deteceted.

## Expected behaviour

worker_1 | [9] [2021-11-05T11:01:40.418] INFO -- : [CIUWO55PWK] Started processing UnqueueMessageJob job
worker_1 | [9] [2021-11-05T11:01:40.426] INFO -- : [CIUWO55PWK] Lock acquired for queued message 20243
worker_1 | [9] [2021-11-05T11:01:40.435] INFO -- : [CIUWO55PWK] Found 0 associated messages to process at the same time (batch key: outgoing-server.example)
worker_1 | [9] [2021-11-05T11:01:40.437] INFO -- : [CIUWO55PWK] [2::1378 20243] Got queued message with exclusive lock
worker_1 | [9] [2021-11-05T11:01:40.965] INFO -- : [O1CLVXMP] Connected to 194.xxx.xx.xx:25 (server.example)
worker_1 | [9] [2021-11-05T11:01:40.968] INFO -- : [O1CLVXMP] Sending message 2::1378 to email@email.com
worker_1 | [9] [2021-11-05T11:01:41.434] INFO -- : [O1CLVXMP] Net::SMTPFatalError: 550-spam detected. transport denied. CT:Bulk
worker_1 | 550-SendFromACL:email@email.com
worker_1 | 550-SendFrom:email@email.com HeaderFrom:NAME
worker_1 | 550-<noreply@email@com> Subject: Example
worker_1 | 550-LegislaÃ§Ã£o Recipients:email@email.com
worker_1 | 550-SIP:176.xxx.xx.xx GateWay:postal.server.com
worker_1 | 550-Connection-ID:20211105110140Z58101mr1.example.com
worker_1 | 550-Mime-Message-ID:<d07376997d00c3304ba3ce49be36d1ee@email.com>
worker_1 | 550 X-CTCH-RefID:str=0001.0A782F28.61850F15.0061,ss=3,re=0.000,recu=0.000,reip=0.000,cl=3,cld=1,fgs=0
worker_1 |
worker_1 | [9] [2021-11-05T11:01:41.488] INFO -- : [CIUWO55PWK] [2::1378 20243] Internal error: Mysql2::Error: Data too long for column 'output' at row 1
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:131:in `_query'
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:131:in `block in query'
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:130:in `handle_interrupt'
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:130:in `query'
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/message_db/database.rb:319:in `query_on_connection'
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/message_db/database.rb:224:in `block in insert'
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/message_db/mysql.rb:20:in `client'
worker_1 | [9] [2021-11-05T11:01:41.489] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/message_db/database.rb:342:in `with_mysql'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/message_db/database.rb:223:in `insert'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/message_db/delivery.rb:8:in `create'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/message_db/message.rb:128:in `create_delivery'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/app/jobs/unqueue_message_job.rb:419:in `block in perform'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/app/jobs/unqueue_message_job.rb:30:in `each'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/app/jobs/unqueue_message_job.rb:30:in `perform'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/worker.rb:55:in `receive_job'
worker_1 | [9] [2021-11-05T11:01:41.490] INFO -- : [CIUWO55PWK] [2::1378 20243] /opt/postal/app/lib/postal/worker.rb:88:in `block in join_queue'
worker_1 | [9] [2021-11-05T11:01:41.491] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/consumer.rb:56:in `call'
worker_1 | [9] [2021-11-05T11:01:41.491] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/channel.rb:1768:in `block in handle_frameset'
worker_1 | [9] [2021-11-05T11:01:41.491] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/consumer_work_pool.rb:108:in `block (2 levels) in run_loop'
worker_1 | [9] [2021-11-05T11:01:41.491] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/consumer_work_pool.rb:103:in `loop'
worker_1 | [9] [2021-11-05T11:01:41.491] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/consumer_work_pool.rb:103:in `block in run_loop'
worker_1 | [9] [2021-11-05T11:01:41.491] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/consumer_work_pool.rb:102:in `catch'
worker_1 | [9] [2021-11-05T11:01:41.492] INFO -- : [CIUWO55PWK] [2::1378 20243] /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/consumer_work_pool.rb:102:in `run_loop'
worker_1 | [9] [2021-11-05T11:01:41.495] INFO -- : [CIUWO55PWK] [2::1378 20243] Queued message was unlocked
worker_1 | [9] [2021-11-05T11:01:41.500] INFO -- : [O1CLVXMP] Finishing up

## Environment details 

 - OS: 18.04.5 LTS
 - Version 2.0.0

## Additional information/context

The solution for that was increasing postal-server-1 and postal-server-2 table deliveries column "output" to greater value.

docker exec -it postal-mariadb bash
mysql -p
use postal-server-1;
ALTER TABLE deliveries MODIFY COLUMN output varchar(2048);
use postal-server-2;
ALTER TABLE deliveries MODIFY COLUMN output varchar(2048);

**Comment by willpower232** [comment]
Interesting this hasn't come up before, I wonder if the receiving server output a whole stack trace or something like that which would be far too long to store in a small field.

Looking at the migrations, the current size is 512. Personally, anything bigger than the common varchar size 255 I would have used a text type and been done with it.

Either way, a migration should be created in https://github.com/postalserver/postal/tree/main/lib/postal/message_db/migrations

**Comment by adamcooke** [comment]
I think I'd rather just truncate the input going into that field. I can't see any huge value in storing outputs larger than 512 characters here.

**Comment by willpower232** [comment]
fair enough, presumably it is possible to either trim the string going in or at least catch the exception so it doesn't derail that Postal  process?

**Comment by dtrckd** [comment]
Hi there, 

I have encountered this issue since yesterday. For unknow reason, Gmail started to block my email an returns an repeated message that says so (from the postal logs)  : 
  
```txt
Our system has detected an unusual amount
421-4.7.28 of unsolicited mail originating from your IP address. To protect our
421-4.7.28 users from spam, mail sent from your IP address h as been temporarily
421-4.7.28 blocked. Please visit
421-4.7.28 https://support.google.com/mail/answer/81126 to review our Bulk
421 4.7.28 Email S enders Guidelines. r68-20020a1c2b47000000b003bd5cc8128dsi2633138wmr.17 - gsmtp
", "421-4.7.28 [2001:41d0:401:3200::3be6] Our system has detected an u nusual amount
421-4.7.28 of unsolicited mail originating from your IP address. To protect our
421-4.7.28 users from spam, mail sent from your IP add ress has been temporarily
421-4.7.28 blocked. Please visit
421-4.7.28 https://support.google.com/mail/answer/81126 to review our Bulk
421 4.7.28 E mail Senders Guidelines. g12-20020a5d64ec000000b0022e669cf2f3si4115062wri.47 - gsmtp
", "421-4.7.28 [2001:41d0:401:3200::3be6] Our system has detecte d an unusual amount
421-4.7.28 of unsolicited mail originating from your IP address. To protect our
421-4.7.28 users from spam, mail sent from your IP address has been temporarily
421-4.7.28 blocked. Please visit
421-4.7.28 https://support.google.com/mail/answer/81126 to review our Bulk
421 4. 7.28 Email Senders Guidelines. c11-20020a05640227cb00b0045c3f5b458fsi13557ede.397 - gsmtp
", "421-4.7.28 [2001:41d0:401:3200::3be6] Our system has de tected an unusual amount
421-4.7.28 of unsolicited mail originating from your IP address. To protect our
421-4.7.28 users from spam, mail sent from your IP address has been temporarily

AN THIS AGAIN x7 TIMES ....
```

Unfortunately, I have lost postal logs around, but errror, was the same as the OP, and now it seems that I am no more banned right,, but as the reason of this are a still a mystery,  guess that could happen again

Postal rendered the error in the ui like so:
![image](https://user-images.githubusercontent.com/8218696/200076776-4dd49737-3eea-4b50-9b9c-572afc729aa4.png)

**Comment by dtrckd** [comment]
And here is a the log, from the worker container:

```
[9] [2022-11-05T01:16:03.138] INFO -- : [VVCC7ZVK] Cannot connect to 74.125.200.27:25 (alt4.gmail-smtp-in.l.google.com) (Net::SMTPServerBusy: 421-4.7.
28 [2001:41d0:401:3200::3be6] Our system has detected an unusual amount                                                                               
421-4.7.28 of unsolicited mail originating from your IP address. To protect our                                                                       
421-4.7.28 users from spam, mail sent from your IP address has been temporarily                                                                       
421-4.7.28 blocked. Please visit                                                                                                                      
421-4.7.28  https://support.google.com/mail/answer/81126 to review our Bulk                                                                           
421 4.7.28 Email Senders Guidelines. t14-20020a1709028c8e00b001783bc48eb2si1182872plo.4 - gsmtp                               
)                                                                                                                                                     
[9] [2022-11-05T01:16:03.138] INFO -- : [VVCC7ZVK] -> No SMTP server available for gmail.com                                            
[9] [2022-11-05T01:16:03.138] INFO -- : [VVCC7ZVK] -> Hostnames: ["gmail-smtp-in.l.google.com", "alt1.gmail-smtp-in.l.google.com", "alt2.gmail-smtp-in
.l.google.com", "alt3.gmail-smtp-in.l.google.com", "alt4.gmail-smtp-in.l.google.com"]                                                             
[9] [2022-11-05T01:16:03.138] INFO -- : [VVCC7ZVK] -> Errors: ["421-4.7.28 [2001:41d0:401:3200::3be6] Our system has detected an unusual amount\n421-4
.7.28 of unsolicited mail originating from your IP address. To protect our\n421-4.7.28 users from spam, mail sent from your IP address has been tempor
arily\n421-4.7.28 blocked. Please visit\n421-4.7.28  https://support.google.com/mail/answer/81126 to review our Bulk\n421 4.7.28 Email Senders Guideli
nes. bq21-20020a5d5a15000000b0023078f98bb5si66224
[.....x7 times.....]
[9] [2022-11-05T01:16:03.139] INFO -- : [KLEGEWCYYS] [1::53 56] Internal error: Mysql2::Error: Data too long for column 'output' at row 1
```

**Comment by catphish** [comment]
This should be fixed by a188a161cbdcfd70158b09b53cef622842357c26

**Comment by audig** [comment]
Hello,
I'm having this issue in postal 2.1.4.

In logs I have the same issue with google, but i consider as normal cause i didn't create SPF and DKIM entry.

But I get an error in Postal UI:
```
Mysql2::Error: Data too long for column 'output' at row 1

Support Ref: J-C93XL7RIWJ
```
